src/server/game/World/IWorld.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "IWorld.h"
+#include "Unit.h"
+
+class IWorld;
```

Comment:
What is IWorld and why cant this be in world.cpp?

Reply:
we have the the world split up into world.h and iworld.h. I tried to consolidated to just the world and essential was met with issues. Unit.cpp was able to be done fine, but the spellauraeffect.cpp required it to be in the iworld.h which i cant for the freaking love of everything i could not understand why as that was giving me a constant compile issue on my local. So then i went ahead with a few more adjustments and it ended up compiling fine, but then the workflow kept crying about IWorld:: being undefined, so i drop a class IWorld in, then i was getting a dam Incomplete type in nested name specifier, then i had to deal with a damn forward declaration annoyance. til finally we got where we are currently that still continues to work as intended. Also the IWorld.cpp had to be created as the damn virtual void AddDelayedDamage was constantly coming up as undefined.